src/drivers/display/renesas_glcdc/lv_renesas_glcdc.c

```diff
@@ -0,0 +1,213 @@
+#else
+#endif
+    }
+    else if(DISPLAY_EVENT_GR1_UNDERFLOW == p_args->event) {
+        __BKPT(0); //Layer 1 Underrun
+    }
+    else if(DISPLAY_EVENT_GR2_UNDERFLOW == p_args->event) {
+        __BKPT(0); //Layer 2 Underrun
+    }
+    else { //DISPLAY_EVENT_FRAME_END
+        __BKPT(0);
+    }
+
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+static void glcdc_init(void)
+{
+    /* Fill the Frame buffer with a colour, to zero out info from previous execution runs */
+    uint32_t count;
+    uint16_t * p = (uint16_t *)&fb_background[0][0];
+
+    for(count = 0; count < sizeof(fb_background) / 2; count++) {
+        *p++ = RGB_565_BLACK;
+    }
+
+    /* Initalize GLCDC driver */
+    uint8_t * p_fb = &fb_background[1][0];
+    fsp_err_t err;
+
+    err = R_GLCDC_Open(&g_display0_ctrl, &g_display0_cfg);
+    if(FSP_SUCCESS != err) {
+        __BKPT(0); //TODO: Better error handling
+    }
+
+    err = R_GLCDC_Start(&g_display0_ctrl);
+    if(FSP_SUCCESS != err) {
+        __BKPT(0); //TODO: Better error handling
+    }
+
+    do {
+        err =
+            R_GLCDC_BufferChange(&g_display0_ctrl,
+                                 (uint8_t *) p_fb,
+                                 (display_frame_layer_t) 0);
+    } while(FSP_ERR_INVALID_UPDATE_TIMING == err);
+
+    /* Enable the backlight */
+    R_IOPORT_PinWrite(&g_ioport_ctrl, DISP_BLEN, BSP_IO_LEVEL_HIGH);
+}
+
+static void flush_direct(lv_display_t * display, const lv_area_t * area, uint8_t * px_map)
+{
+    FSP_PARAMETER_NOT_USED(area);
+    //Display the frame buffer pointed by px_map
+
+    if(!lv_display_flush_is_last(display)) return;
+#if defined(RENESAS_CORTEX_M85)
+#if (BSP_CFG_DCACHE_ENABLED)
+    int32_t size;
+    /* Invalidate cache - so the HW can access any data written by the CPU */
+    if(display->render_mode == LV_DISPLAY_RENDER_MODE_PARTIAL) {
+        size = lv_area_get_width(area) * lv_area_get_height(area) * lv_color_format_get_size(display->color_format);
+    }
+    else {
+        size = sizeof(fb_background[0]);
+    }
+    SCB_CleanInvalidateDCache_by_Addr(px_map, size);
+#endif
+#endif
+
+    R_GLCDC_BufferChange(&g_display0_ctrl,
+                         (uint8_t *) px_map,
+                         (display_frame_layer_t) 0);
+}
+
+static void flush_wait_direct(lv_display_t * display)
+{
+    if(!lv_display_flush_is_last(display)) return;
+
+#if BSP_CFG_RTOS == 2              // FreeRTOS
+    //
+    // If Vsync semaphore has already been set, clear it then wait to avoid tearing
+    //
+    if(uxSemaphoreGetCount(_SemaphoreVsync)) {
+        xSemaphoreTake(_SemaphoreVsync, 10);
+    }
+
+    xSemaphoreTake(_SemaphoreVsync, portMAX_DELAY);
+#endif
+
+}
+
+static void flush_partial(lv_display_t * display, const lv_area_t * area, uint8_t * px_map)
+{
+    LV_UNUSED(display);
+
+    int32_t w = lv_area_get_width(area);
+    int32_t h = lv_area_get_height(area);
+    SCB_CleanInvalidateDCache_by_Addr(px_map, w * h * 2);
```

Comment:
It should be called in line 204 for the frame buffer.
```c
SCB_CleanInvalidateDCache_by_Addr(fb, DISPLAY_HSIZE_INPUT0 * DISPLAY_VSIZE_INPUT0 * 2);
```

Reply:
replacing 191 or keep both? and why call for the whole buffer in partial mode?

Reply:
Only at line 191. 
It would be more optimal to invalidate on the changed area. It can be added now or in a later PR as an optimization.

Reply:
The suggested line does not work, as `fb` is being incremented in the loop, and pointing to a bad location by line 204.
This works* at the end of function:
```c
SCB_CleanInvalidateDCache_by_Addr(fb_background[1], sizeof(fb_background[1]));
```
Or this in the loop:
```c
SCB_CleanInvalidateDCache_by_Addr(fb, w * 2);
```

However it only works for the artifacts seen at switch transition. Pressing the "Log out" button also creates artifacts and they are not resolved by cache invalidation. It might be a different issue, as the stripes are inside the button, with white color. At no point of the animation should that are be white, so it's not a "slow transition" related to cache emptying out.

This is not present in direct mode, however a scroll bar appears after pressing the button once. The bar does not appear in the PC emulator with matching screen resolution.